### PR TITLE
Block pool creation testing - cypress

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts [hunk 1]
```diff
@@ -1,9 +1,61 @@
...
+const poolMessage: {
+  [key in POOL_PROGRESS]?: () => string;
+} = {
+  [POOL_PROGRESS.FAILED]: () => `Pool "${poolName}" already exists`,
+  [POOL_PROGRESS.CREATED]: () => `Pool ${poolName} was successfully created`,
+};
+
+const storagePool = {
+  prepareStorageClassForm: () => {
+    // Select provisioner
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:23]: Moved these part from multiple-pool.ts to here, and i delete views/multiple-pool.ts (since it is specific code to pool modal, i moved here)

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  // deviceType
+  cy.byTestID('volume-type-dropdown').click();
+  cy.byTestID(volumeType).click();
+  // compression
+  cy.byTestID('compression-checkbox').check();
+};
+
+export const createBlockPool = () => {
+  commonFlows.navigateToOCS();
+  cy.byLegacyTestID('horizontal-link-Block Pools').click();
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:27]: planned to reuse this create function for delete and update block pool testcases

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  cy.exec(`oc get cephBlockPool ${poolName} -n openshift-storage -o json`).then((res) => {
+    const blockPool = JSON.parse(res.stdout);
+    expect(blockPool.spec?.replicated?.size).toEqual(Number(replica));
+    expect(blockPool.spec?.compressionMode).toEqual(compressionEnabled ? 'aggressive' : '');
+    expect(blockPool.spec?.parameters?.compression_mode).toEqual(
+      compressionEnabled ? 'aggressive' : '',
+    );
+  });
+
+export const createBlockPool = (alreadyExists: boolean = false) => {
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:83]: create block pool from block pool page, and it will be re-used by update and delete testcases

**Comment by afreen23** [comment]
Merge after #8924

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/create-block-pool.spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  after(() => {
+    checkErrors();
+    cy.logout();
+  });
+
+  it('Check for a new pool creation', () => {
+    createBlockPool();
+
+    cy.log('Try to create a new pool with already existing name');
+    createBlockPool(true);
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/create-block-pool.spec.ts:20]: Yes, it makes sense. Ne need this check

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+  cy.byTestID('dropdown-menu-item-link').click();
+
+  // select block pool
+  cy.byTestID(storagePoolDropdown).click();
+  cy.byTestID('create-new-pool-button').click();
+};
+
+const createBlockPool = (poolCreationAction: string) => {
+  // Make sure the storage pool creation form is open
+  modal.modalTitleShouldContain('Create New Block Pool');
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:29]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts [hunk 6]
```diff
@@ -1,9 +1,42 @@
...
+  cy.byTestID('dropdown-menu-item-link').should('contain', 'openshift-storage.rbd.csi.ceph.com');
+  cy.byTestID('dropdown-menu-item-link').click();
+
+  // select block pool
+  cy.byTestID(storagePoolDropdown).click();
+  cy.byTestID('create-new-pool-button').click();
+};
+
+const createBlockPool = (poolCreationAction: string) => {
+  // Make sure the storage pool creation form is open
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:28]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  navigateToBlockPool();
+  cy.byTestID('item-create').click();
+  blockPoolFooter('create');
+  populateBlockPoolform();
+  blockPoolFooter('create', true);
+
+  if (!alreadyExists) {
+    // verify pool creation
+    cy.byTestID('status-text').contains('Ready');
+    verifyBlockPool();
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:93]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+export const createBlockPool = (alreadyExists: boolean = false) => {
+  navigateToBlockPool();
+  cy.byTestID('item-create').click();
+  blockPoolFooter('create');
+  populateBlockPoolform();
+  blockPoolFooter('create', true);
+
+  if (!alreadyExists) {
+    // verify pool creation
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:91]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts [hunk 9]
```diff
@@ -1,9 +1,42 @@
...
+  verifyBlockPool,
+} from '../views/block-pool';
 
 // Pool var
-const poolName: string = 'example.pool';
-const replicaCount: string = '2';
+const storagePoolDropdown: string = 'pool-dropdown-toggle';
+
+const prepareStorageClassForm = () => {
+  // select provisioner
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:16]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+  [POOL_PROGRESS.NOTALLOWED]:
+    "Pool management tasks are not supported for default pool and Openshift Container Storage's external mode.",
+};
+
+export const navigateToBlockPool = () => {
+  commonFlows.navigateToOCS();
+  cy.byLegacyTestID('horizontal-link-Block Pools').click();
+};
+
+export const populateBlockPoolform = () => {
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:24]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts]: BlockPool is not a resource, it is a display name. CephBlockPool is the resource name.

In that case, i have change this to `Create block pool` 

including from OCS operator to display header

[the remark this answers is not in the document]
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts]: I checked with core team, now everything changed to no space. So instead of `block pool` or `Block Pool`.
We have ti say `BlockPool`.
Please change this in `label` and `labelPlural` of `CephBlockPool` model as well.

File: frontend/public/components/utils/button-bar.jsx [hunk 11]
```diff
@@ -25,6 +25,7 @@ const ErrorMessage = ({ message }) => {
       className="co-alert co-alert--scrollable"
       variant="danger"
       title={t('public~An error occurred')}
+      data-test="error-message-alert"
```
Comment (afreen23) [inline frontend/public/components/utils/button-bar.jsx:28]: @spadgett PTAL ^^

**Comment by GowthamShanmugam** [comment]
@spadgett Please review

**Review by bipuladh — CHANGES_REQUESTED** [review]
`BlockPool` naming convention changes should not go in this PR. Create a separate PR for that. I think it could be a BZ. This is related to testing.

File: frontend/public/components/utils/button-bar.jsx (hunk 11 above)
Comment (spadgett) [reply frontend/public/components/utils/button-bar.jsx:28]: Hi @afreen23. Maybe I'm missing it, but I don't see where this is used in the tests. Can you point out the place?

Comment (GowthamShanmugam) [reply frontend/public/components/utils/button-bar.jsx:28]: Ah! i got it, Initially there was a testing to check pool already existing error message, i think i added to check error message from button bar

Comment (GowthamShanmugam) [reply frontend/public/components/utils/button-bar.jsx:28]: My bad i will remove

Comment (afreen23) [reply frontend/public/components/utils/button-bar.jsx:28]: Yes,  found that this was only  required herehttps://github.com/openshift/console/pull/8657#discussion_r635198910 . 
Surely we can leave this untouched. Sorry for the lame ping !  :facepalm:

File: frontend/packages/ceph-storage-plugin/src/models.ts [hunk 12]
```diff
@@ -27,8 +27,8 @@ export const OCSServiceModel: K8sKind = {
 };
 
 export const CephBlockPoolModel: K8sKind = {
-  label: 'Block Pool',
-  labelPlural: 'Block Pools',
+  label: 'BlockPool',
+  labelPlural: 'BlockPools',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/models.ts:31]: Do it in a different PR.

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 13]
```diff
@@ -323,7 +323,7 @@ const plugin: Plugin<ConsumedExtensions> = [
     properties: {
       id: 'edit',
       kind: models.CephBlockPoolModel.kind,
-      label: '%ceph-storage-plugin~Edit Block Pool%',
+      label: '%ceph-storage-plugin~Edit BlockPool%',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts:326]: Do it in a different PR.

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 14]
```diff
@@ -763,7 +763,7 @@ const plugin: Plugin<ConsumedExtensions> = [
     properties: {
       id: 'delete',
       kind: models.CephBlockPoolModel.kind,
-      label: '%ceph-storage-plugin~Delete Block Pool%',
+      label: '%ceph-storage-plugin~Delete BlockPool%',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts:766]: Do it in a different PR.

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx [hunk 15]
```diff
@@ -40,9 +40,9 @@ const UpdateBlockPoolModal = withHandlePromise((props: UpdateBlockPoolModalProps
   );
   const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
 
-  const MODAL_TITLE = t('ceph-storage-plugin~Edit Block Pool');
+  const MODAL_TITLE = t('ceph-storage-plugin~Edit BlockPool');
   const MODAL_DESC = t(
-    'ceph-storage-plugin~A block pool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency.',
+    'ceph-storage-plugin~A BlockPool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency.',
   );
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx:46]: In a different PR.

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx [hunk 16]
```diff
@@ -160,7 +160,7 @@ export const BlockPoolModalFooter = (props: BlockPoolModalFooterProps) => {
             variant={buttonProp.variant}
             isDisabled={buttonProp.disable}
             id={buttonProp.id}
-            data-test={buttonProp.id}
+            data-test-id={buttonProp.id}
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx:163]: `data-test` is the preferred method. Why use `id`?

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx [hunk 17]
```diff
@@ -136,7 +136,7 @@ const DeleteBlockPoolModal = withHandlePromise((props: DeleteBlockPoolModalProps
         <StatusBox
           loaded={isLoaded && pvcLoaded && scLoaded}
           loadError={loadError ?? pvcLoadError ?? scLoadError}
-          label={t('ceph-storage-plugin~Block Pool Delete Modal')}
+          label={t('ceph-storage-plugin~BlockPool Delete Modal')}
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx:139]: In a different PR.

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx [hunk 18]
```diff
@@ -86,7 +86,7 @@ const DeleteBlockPoolModal = withHandlePromise((props: DeleteBlockPoolModalProps
     handlePromise(k8sKill(CephBlockPoolModel, blockPoolConfig), () => close());
   };
 
-  const MODAL_TITLE = t('ceph-storage-plugin~Delete Block Pool');
+  const MODAL_TITLE = t('ceph-storage-plugin~Delete BlockPool');
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx:89]: In a different PR.

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx [hunk 19]
```diff
@@ -43,9 +43,9 @@ export const CreateBlockPoolModal = withHandlePromise((props: CreateBlockPoolMod
   const [timer, setTimer] = React.useState<NodeJS.Timer>(null);
 
   const MODAL_DESC = t(
-    'ceph-storage-plugin~A block pool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency.',
+    'ceph-storage-plugin~A BlockPool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency.',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx:46]: In a different PR.

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx [hunk 20]
```diff
@@ -43,9 +43,9 @@ export const CreateBlockPoolModal = withHandlePromise((props: CreateBlockPoolMod
...
   );
-  const MODAL_TITLE = t('ceph-storage-plugin~Create New Block Pool');
+  const MODAL_TITLE = t('ceph-storage-plugin~Create BlockPool');
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx:48]: In a different PR.

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx (hunk 16 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx:163]: integration-tests-cypress/views/modal.ts is still using byLegacyTestID (cy.byLegacyTestID('modal-cancel-action').should('not.exist')).

same pool creation form is used between block pool create a page under OCS operator and block pool create modal under SC. So I am creating a footer manually, just to use this cypress views/modal.ts functions i am keeping it as legacy id

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+import { commonFlows } from './common';
+import { POOL_PROGRESS } from '../../src/constants/storage-pool-const';
+
+// Pool var
+export const poolName: string = 'example.pool';
+export const replicaCount: string = '2';
+export const volumeType: string = 'ssd';
+
+export const poolMessage: {
+  [key in POOL_PROGRESS]?: string;
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:11]: POOL_PROGRESS has a lot of statuses, i am not added all here, so it should be optional

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:30]: Let's create a different PR for fixing this.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts [hunk 22]
```diff
@@ -1,9 +1,43 @@
...
-const poolName: string = 'example.pool';
-const replicaCount: string = '2';
+const storagePoolDropdown: string = 'pool-dropdown-toggle';
+
+const prepareStorageClassForm = () => {
+  cy.log('Selecting provisioner');
+  cy.byTestID('storage-class-provisioner-dropdown').click();
+  cy.byLegacyTestID('dropdown-text-filter').type('openshift-storage.rbd.csi.ceph.com');
+  cy.byTestID('dropdown-menu-item-link').should('contain', 'openshift-storage.rbd.csi.ceph.com');
+  cy.byTestID('dropdown-menu-item-link').click();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:20]:
```suggestion
  cy.byTestID('dropdown-menu-item-link').contains('openshift-storage.rbd.csi.ceph.com').click();
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts [hunk 23]
```diff
@@ -1,9 +1,43 @@
...
+  populateBlockPoolForm,
+  blockPoolFooter,
+  deleteBlockPoolFromCli,
+  verifyBlockPoolYaml,
+} from '../views/block-pool';
 
 // Pool var
-const poolName: string = 'example.pool';
-const replicaCount: string = '2';
+const storagePoolDropdown: string = 'pool-dropdown-toggle';
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts:13]: Move this to the corresponding view file.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+            .should('be.visible')
+            .click()
+        : cy
+            .byLegacyTestID('confirm-action')
+            .scrollIntoView()
+            .should('be.disabled');
+  }
+};
+
+export const verifyBlockPoolYaml = (
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:77]:
```suggestion
export const verifyBlockPoolJSON = (
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+  compressionEnabled: boolean = true,
+  replica: string = replicaCount,
+) =>
+  cy.exec(`oc get cephBlockPool ${poolName} -n openshift-storage -o json`).then((res) => {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:81]:
```suggestion
  cy.exec(`oc get cephBlockPool ${poolName} -n ${NS} -o json`).then((res) => {
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+export const poolMessage: {
+  [key in POOL_PROGRESS]?: string;
+} = {
+  [POOL_PROGRESS.FAILED]: `Pool "${poolName}" already exists`,
+  [POOL_PROGRESS.CREATED]: `Pool ${poolName} was successfully created`,
+  [POOL_PROGRESS.NOTALLOWED]:
+    "Pool management tasks are not supported for default pool and Openshift Container Storage's external mode.",
+};
+
+export const navigateToBlockPool = () => {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:19]:
```suggestion
const navigateToBlockPool = () => {
```
I don't see it being to used anywhere else.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 27]
```diff
@@ -0,0 +1,105 @@
...
+
+export const createBlockPool = () => {
+  navigateToBlockPool();
+  cy.byTestID('item-create').click();
+  blockPoolFooter('create');
+  populateBlockPoolForm();
+  blockPoolFooter('create', true);
+  cy.log('Verify a new block pool creation');
+  cy.byTestID('status-text').contains('Ready');
+  verifyBlockPoolYaml();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:98]:
```suggestion
  verifyBlockPoolJSONl();
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 28]
```diff
@@ -0,0 +1,105 @@
...
+      visible
+        ? cy
+            .byLegacyTestID('confirm-action')
+            .scrollIntoView()
+            .should('be.visible')
+            .click()
+        : cy
+            .byLegacyTestID('confirm-action')
+            .scrollIntoView()
+            .should('be.disabled');
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:73]:
```suggestion
break;
```
Just use `modal.submitShouldBeDisabled` and `modal.submitShouldBeEnabled` where necessary( prolly before and after filling the fields). Performing multiple tests from a function just makes understanding the code more difficult.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 29]
```diff
@@ -0,0 +1,105 @@
...
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.CREATED]);
+      cy.byLegacyTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
+      break;
+    case Actions.notAllowed:
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.NOTALLOWED]);
+      cy.byLegacyTestID('modal-close-action')
+        .should('be.visible')
+        .click();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:61]:
```suggestion
      cy.byLegacyTestID('modal-close-action').click();
```
Commands perform certain validation before they run hence the `visible` assertion is not required.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+      cy.byLegacyTestID('modal-try-again-action').should('be.visible');
+      cy.byLegacyTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
+      break;
+    case Actions.created:
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.CREATED]);
+      cy.byLegacyTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:55]:
```suggestion
      cy.byLegacyTestID('modal-finish-action').click();
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 31]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+export const blockPoolFooter = (action: string, visible: boolean = false) => {
+  switch (action) {
+    case Actions.failed:
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.FAILED]);
+      cy.byLegacyTestID('modal-try-again-action').should('be.visible');
+      cy.byLegacyTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:49]:
```suggestion
      modal.submit();
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 32]
```diff
@@ -0,0 +1,105 @@
...
+
+export const navigateToBlockPool = () => {
+  commonFlows.navigateToOCS();
+  cy.byLegacyTestID('horizontal-link-Block Pools').click();
+};
+
+export const populateBlockPoolForm = () => {
+  cy.byTestID('new-pool-name-textbox').type(poolName);
+  cy.byTestID('replica-dropdown').click();
+  cy.byLegacyTestID(replicaCount)
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:27]: I do not like this approach where the data-test-id is a value instead of a proper identifier. Values are rendered as part of html. data-test-id should be uniquely identifying that particular field. Wouldn't using `contains` followed by `last` work for this scenario?

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+    .last()
+    .contains(`${replicaCount}-way Replication`)
+    .click();
+  cy.byTestID('volume-type-dropdown').click();
+  cy.byTestID(volumeType).click();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:32]:
```suggestion
  cy.contains(volumeType).click();
```
Wouldn't this work?

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts (hunk 32 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts:27]: ack, let me check how to change this

**Review by bipuladh — APPROVED** [review]
/lgtm
